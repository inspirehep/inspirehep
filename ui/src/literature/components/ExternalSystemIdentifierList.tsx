import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { List } from 'immutable';

import InlineList from '../../common/components/InlineList';

class ExternalSystemIdentifierList extends Component {
  render() {
    const { externalSystemIdentifiers } = this.props;
    return (
      <InlineList
        label="View in"
        items={externalSystemIdentifiers}
        extractKey={esid => esid.get('url_link')}
        renderItem={esid => (
          <a href={esid.get('url_link')}>{esid.get('url_name')}</a>
        )}
      />
    );
  }
}

ExternalSystemIdentifierList.propTypes = {
  externalSystemIdentifiers: PropTypes.instanceOf(List),
};

ExternalSystemIdentifierList.defaultProps = {
  externalSystemIdentifiers: null,
};

export default ExternalSystemIdentifierList;
