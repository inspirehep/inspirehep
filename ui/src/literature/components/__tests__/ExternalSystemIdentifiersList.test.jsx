import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import ExternalSystemIdentifierList from '../ExternalSystemIdentifierList';

describe('ExternalSystemIdentifierList', () => {
  it('renders with external system identifiers', () => {
    const externalSystemIdentifiers = fromJS([
      {
        url_link: 'https://cds.cern.ch/record/12345',
        url_name: 'CERN Document Server',
      },
      {
        url_link: 'https://ui.adsabs.harvard.edu/abs/123.1234',
        url_name: 'ADS Abstract Service',
      },
    ]);
    const wrapper = shallow(
      <ExternalSystemIdentifierList
        externalSystemIdentifiers={externalSystemIdentifiers}
      />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
