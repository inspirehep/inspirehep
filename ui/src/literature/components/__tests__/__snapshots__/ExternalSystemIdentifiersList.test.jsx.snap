// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalSystemIdentifierList renders with external system identifiers 1`] = `
<div
  className="__InlineList__"
>
  <span
    className=""
  >
    View in
    : 
  </span>
  <ul>
    <li
      key="https://cds.cern.ch/record/12345-0"
    >
      <a
        href="https://cds.cern.ch/record/12345"
      >
        CERN Document Server
      </a>
      , 
    </li>
    <li
      key="https://ui.adsabs.harvard.edu/abs/123.1234-2"
    >
      <a
        href="https://ui.adsabs.harvard.edu/abs/123.1234"
      >
        ADS Abstract Service
      </a>
    </li>
  </ul>
</div>
`;
